app/javascript/react/src/components/App.jsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import store from '../redux/store';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import RandomMessage from './Greeting.jsx';
+
+const App = () => {
+  return (
+    <Routes>
+      <Route path="/" element={<RandomMessage />} />
+    </Routes>
+  );
+};
+
+const root = ReactDOM.createRoot(document.getElementById('root'))
+root.render(
+  <React.StrictMode>
+    <Provider store={store}>
+      <Router>
+        <App />
+      </Router>
+    </Provider>
+  </React.StrictMode>
+);
```

Comment:
- You did well so far, and your app is close to working perfectly, you have not configured `rack-cors` in your application hence I cannot see any displayed greeting on the UI. Please add the gem 'rack-cors' to your gemfile. You might find this [guide](https://medium.com/nerd-for-tech/rails-rack-cors-6c0c7ee438bb) helpful. Please fix this 👍 👍